tools/i18n/unescape-contents

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+import argparse
+import html
+import json
+import os
+import sys
+from contextlib import contextmanager
+from pathlib import PurePath
+from typing import IO, Any, Generator, NamedTuple, Union
+
+
+class CLIArgs(NamedTuple):
+    filename: str
+    indent_level: int
+
+
+def parse_args() -> CLIArgs:
+    parser = argparse.ArgumentParser(
+        prog="unescape-contents",
+        description="Unescape Unicode and HTML entities in a JSON file",
+    )
+
+    parser.add_argument("filename", type=str)
+    parser.add_argument("--indent-level", dest="indent_level", type=int, default=2, required=False)
+    args = parser.parse_args()
+
+    return CLIArgs(args.filename, args.indent_level)
+
+
+@contextmanager
+def tempfile_for_atomic_overwrite(
+    destination: Union[str, "os.PathLike[str]"]
+) -> Generator[IO[Any], None, None]:
+    """
+    Creates a sibling temporary file in the same directory as `destination` and
+    yields the file-like object returned by `open()`. Upon completion of the
+    context block, the file handle is closed and the temporary file is removed,
+    either by replacing `destination` with its contents (see `os.replace()`
+    documentation) if the context succeeded, or by `os.unlink()`-ing it if the
+    context failed.
+
+    This is not a general-purpose temporary file creator: for usecases that
+    don't require moving the temporary file instead of deleting it, use the
+    `tempfile` module.
+    """
+
+    source_path = PurePath(os.path.abspath(destination))
+    temp_path = source_path.with_name(f".{source_path.name}.unescaped")
+    temp_file = open(temp_path, "w+")
+
+    try:
+        yield temp_file
+        temp_file.close()
+        os.replace(temp_file.name, destination)
+    finally:
+        try:
+            temp_file.close()
+            os.unlink(temp_path)
+        except FileNotFoundError:
+            # In the event of success, we've already moved this file
+            pass
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    print(f"unescaping file {args.filename}", file=sys.stderr)
+
+    with tempfile_for_atomic_overwrite(args.filename) as dest:
+        with open(args.filename) as source:
+            # WARNING: json.load will load the entire file into memory [1], and
+            # thus this should not be used on huge files. Our translation files
+            # are all <1MB in size, however, so in context this should be fine.
+            #
+            # [1] https://pythonspeed.com/articles/json-memory-streaming/
+            json_data = json.load(source)
+
+            for key, value in json_data.items():
+                if key == html.unescape(key):
```

Comment:
I don’t think we can be confident that a heuristic like this is correct and, importantly, will remain correct in the future, until we have a check like [this](https://chat.zulip.org/#narrow/stream/58-translation/topic/removing.20HTML.20escapes.20left.20behind.20by.20Transifex.20MT.20.286.2Ex.29/near/1486145) in place. That check would not be satisfied at present, I think, so this will involve actual changes.